scheduler/pkg/store/experiment/db.go

```diff
@@ -66,10 +66,20 @@ func (edb *ExperimentDBManager) save(experiment *Experiment) error {
 	if err != nil {
 		return err
 	}
-	return edb.db.Update(func(txn *badger.Txn) error {
-		err = txn.Set([]byte(experiment.Name), experimentBytes)
-		return err
-	})
+
+	if !experiment.Deleted {
+		return edb.db.Update(func(txn *badger.Txn) error {
+			err = txn.Set([]byte(experiment.Name), experimentBytes)
+			return err
+		})
+	} else {
+		ttl := utils.DeletedResourceTTL
+		return edb.db.Update(func(txn *badger.Txn) error {
```

Comment:
nit: given that we have a cron now to recycle delete resources from the store we could just do the same thing by just deleting the record using the `delete` helper? therefore it is less changes and feels the logic of recycling resources is contained in one place / logic.

Reply:
It could be done like that, but the date and time of deletiong needs to be persisted somehow. Some sort of longer term refactor of this portion of the code would be ideal though, for example, the in-memory store could be merged with the DB layer and become an implementation detail there. I think this would reduce the coupling between the functions and keep everything contained.

Reply:
ok got it, so currently we rely on the `ExpiresAt()` api that already exists as opposed to persisting a new field `DeletedAt`.

Reply:
Yep... maybe as part of the longer term refactoring, a new field could be persisted. I've got a few different versions of this change on different branches (one with a DB migration, another with the in-memory store and DB merged). I think this would benefit from a slight redesign, but some alignment on the requirements is a prerequisite for that.